code/obj/railing.dm

```diff
@@ -0,0 +1,344 @@
+					is_athletic_jump = 1
+		if (The_Railing)
+			the_railing = The_Railing
+			if (get_dist(ownerMob, the_railing) == 0)
+				jump_target = get_step(the_railing, the_railing.dir)
+			else
+				jump_target = get_turf(the_railing)
+
+	onUpdate()
+		..()
+		// you gotta hold still to jump!
+		if (get_dist(ownerMob, the_railing) > 1)
+			interrupt(INTERRUPT_ALWAYS)
+			ownerMob.show_text("Your jump was interrupted!", "red")
+			return
+
+	onStart()
+		..()
+		if (get_dist(ownerMob, the_railing) > 1 || the_railing == null || ownerMob == null)
+			interrupt(INTERRUPT_ALWAYS)
+			return
+		for(var/mob/O in AIviewers(ownerMob))
+			O.show_text("[ownerMob] begins to pull [himself_or_herself(ownerMob)] over [the_railing].", "red")
+
+	onEnd()
+		..()
+		var/bunp //the name of the thing we have bunp'd into when trying to jump the railing
+		var/list/bunp_whitelist = list(/obj/railing, /obj/decal/stage_edge) // things that we cannot bunp into
+		if (jump_target.density)
+			bunp = jump_target.name
+			no_no_zone = 1
+		else
+			for (var/obj/o in jump_target.contents)
+				for (var/i=1, i<=bunp_whitelist.len+1, i++) //+1 so we can actually go past the whitelist len check
+					if (!(i > bunp_whitelist.len))
+						// if it's an exception to the bunp rule...
+						if (istype (o, bunp_whitelist[i]))
+							break
+
+					// don't proceed just yet if we aren't done going through the whitelist!
+					else if (i <= bunp_whitelist.len)
+						continue
+
+					// otherwise, if it's a dense thing...
+					else if (o.density)
+						bunp = o.name
+						no_no_zone = 1
+						break
+
+		if(no_no_zone)
+			// if they are a living mob, make them TASTE THE PAIN
+			if (istype(ownerMob, /mob/living))
+				if (!ownerMob.hasStatus("weakened"))
+					ownerMob.changeStatus("weakened", 4 SECONDS)
+					playsound(the_railing, 'sound/impact_sounds/Metal_Clang_3.ogg', 50, 1, -1)
+					for(var/mob/O in AIviewers(ownerMob))
+						O.show_text("[ownerMob] tries to climb straight into \the [bunp].[prob(30) ? pick(" What a goof!!", " A silly [ownerMob.name].", " <b>HE HOO HE HA</b>", " Good thing [he_or_she(ownerMob)] didn't bump [his_or_her(ownerMob)] head!") : null]", "red")
+				// HE HE U BUNPED YOUR HEAD
+				if (prob(25))
+					ownerMob.changeStatus("weakened", 4 SECONDS)
+					ownerMob.TakeDamage("head", 0, 10)
+					playsound(the_railing, 'sound/impact_sounds/Metal_Hit_Heavy_1.ogg', 50, 1, -1)
+					for(var/mob/O in AIviewers(ownerMob))
+						O.show_text("[ownerMob] bumps [his_or_her(ownerMob)] head on \the [bunp].[prob(30) ? pick(" Oof, that looked like it hurt!", " Is [he_or_she(ownerMob)] okay?", " Maybe that wasn't the wisest idea...", " Don't do that!") : null]", "red")
+
+			return
+
+		// otherwise, the user jumps over without issue!
+		ownerMob.set_loc(jump_target)
+		for(var/mob/O in AIviewers(ownerMob))
+			var/the_text = null
+			if (is_athletic_jump) // athletic jumps are more athletic!!
+				the_text = "[ownerMob] swooces right over [the_railing]!"
```

Comment:
spelling?

Reply:
The spelling was deliberate. It's a reference to a meme, but I could make it something else!